filebeat/harvester/log.go

```diff
@@ -186,52 +216,38 @@ func (h *Harvester) openFile() error {
 		return err
 	}
 
-	// yay, open file
-	h.file = source.File{f}
 	return nil
 }
 
 func (h *Harvester) initFileOffset(file *os.File) error {
-	offset, err := file.Seek(0, os.SEEK_CUR)
 
-	if h.getOffset() > 0 {
-		// continue from last known offset
-		logp.Debug("harvester",
-			"harvest: %q position:%d (offset snapshot:%d)", h.path, h.getOffset(), offset)
-		_, err = file.Seek(h.getOffset(), os.SEEK_SET)
-	} else if h.config.TailFiles {
-		// tail file if file is new and tail_files config is set
-		logp.Debug("harvester", "harvest: (tailing) %q (offset snapshot:%d)", h.path, offset)
+	var err error
 
-		offset, err = file.Seek(0, os.SEEK_END)
-		h.SetOffset(offset)
+	// continue from last known offset
+	if h.state.Offset > 0 {
+		h.state.Offset, err = file.Seek(h.state.Offset, os.SEEK_SET)
```

Comment:
do not set `h.state.Offset` here. Which value will the offset be if seek fails (even if fail is temporary)?